WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -87,19 +87,41 @@ def index
     end
 
     if @display == 'admin'
-      @competitions = @competitions.includes(:delegates, :delegate_report)
+      @competitions = @competitions.includes(:delegate_report)
     end
 
     unless params[:region] == "all"
-      @competitions = @competitions.select { |competition| competition.belongs_to_region?(params[:region]) }
+      @competitions = @competitions.where("countryId = ? or continentId = ?", params[:region], params[:region])
```

Comment:
`continentId` is on the Countries table and has an index. `countryId` is on Competitions and doesn't so we probably should add one for this.